File: 2466. Count Ways To Build Good Strings/main.go
package main

func countGoodStrings(low int, high int, zero int, one int) int {
	cache := make(map[int]int)

	var getStrings func(length int) int
	getStrings = func(length int) int {
		if length > high {
			return 0
		}

		if cachedVal, exists := cache[length]; exists {
			return cachedVal
		}

		val := 0
		if length >= low && length <= high {
			val = 1
		}

		cache[length] = (val + getStrings(length+one) + getStrings(length+zero)) % 1000000007

		return cache[length]
	}

	return getStrings(0)
}

File: 2466. Count Ways To Build Good Strings/main.py
class Solution:
    def countGoodStrings(self, low: int, high: int, zero: int, one: int) -> int:
        cache = {}

        def get_strings(length):
            if length > high: 
                return 0

            if length in cache:
                return cache[length]    
            
            cache[length] = (
                (1 if low <= length <= high else 0)
                + get_strings(length + zero)
                + get_strings(length + one)
            ) % 1000000007

            return cache[length] 

        return get_strings(0)